src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -51,19 +53,29 @@ protected internal override void Setup(FeatureConfigurationContext context)
                     {Headers.HostDisplayName, hostInfo.DisplayName}
                 };
 
+                var recoveryActionExecutor = new RecoveryActionExecutor(b.Build<IDispatchMessages>(), errorQueue, staticFaultMetadata);
+
                 return new MoveFaultsToErrorQueueBehavior(b.Build<CriticalError>(),
-                    staticFaultMetadata,
                     transportTransactionMode,
                     failureInfoStorage,
-                    b.Build<IDispatchMessages>(),
-                    errorQueue);
+                    recoveryActionExecutor);
             }, "Moves failing messages to the configured error queue");
 
             if (IsDelayedRetriesEnabled(context.Settings))
             {
                 var retryPolicy = GetDelayedRetryPolicy(context.Settings);
 
-                context.Pipeline.Register("SecondLevelRetries", b => new SecondLevelRetriesBehavior(retryPolicy, localAddress, failureInfoStorage), "Performs second level retries");
+                context.Pipeline.Register("SecondLevelRetries", b =>
+                {
+                    var delayedRetryExecutor = new DelayedRetryExecutor(
+                        localAddress,
+                        b.Build<IDispatchMessages>(),
+                        context.DoesTransportSupportConstraint<DelayedDeliveryConstraint>()
+                            ? null
+                            : context.Settings.Get<TimeoutManagerAddressConfiguration>().TransportAddress);
```

Comment:
@andreasohlund I think we need to disable SLR in case the TM is disabled on non-native deferral transports? (which I think is something we still need to fix on the recoverability)